roslibrust/src/ros1/node.rs

```diff
@@ -278,15 +312,19 @@ impl Node {
                 );
             }
             NodeMsg::SetPeerPublishers { topic, publishers } => {
-                let item = self.subscriptions.get_mut(&topic);
-                let mut item = match item {
-                    Some(i) => i,
-                    None => {
-                        log::warn!("Got peer publisher update for topic we weren't subscribed to, ignoring");
-                        return;
+                if let Some(mut subscription) = self.subscriptions.get_mut(&topic) {
+                    for publisher_uri in publishers {
+                        if let Err(err) = subscription.add_publisher_source(&publisher_uri).await {
+                            log::error!(
+                                "Unable to create subscribe stream for topic {topic}: {err}"
+                            );
+                        }
                     }
-                };
-                item.known_publishers = publishers;
+                } else {
+                    log::warn!(
```

Comment:
[nit] maybe log::debug!?

Reply:
I think this condition, while harmless, is indicative of a bug in our client so I'd prefer to keep the severity high.